src/core/bonded_interactions/angle_cosine.hpp

```diff
@@ -46,9 +46,10 @@ int angle_cosine_set_params(int bond_type, double bend, double phi0);
  *  @return Forces on the second, first and third particles, in that order.
  */
 inline std::tuple<Utils::Vector3d, Utils::Vector3d, Utils::Vector3d>
-calc_angle_cosine_3body_forces(Particle const *p_mid, Particle const *p_left,
-                               Particle const *p_right,
-                               Bonded_ia_parameters const *iaparams) {
+calc_angle_cosine_3body_forces(Particle const *const p_mid,
+                               Particle const *const p_left,
+                               Particle const *const p_right,
+                               Bonded_ia_parameters const *const iaparams) {
```

Comment:
And we want to have bond-specific parameter types

Reply:
this means exposing the internal implementation of particles in `pressure_inline.hpp`:
```diff
   switch (iaparams->type) {
   case BONDED_IA_ANGLE_HARMONIC:
     std::tie(f_mid, f_left, f_right) =
-        calc_angle_harmonic_3body_forces(p_mid, p_left, p_right, iaparams);
+        calc_angle_harmonic_3body_forces(p_mid, p_left, p_right, iaparams.p);
     break;
```

Reply:
no, I thought of splitting ```Bonded_ia_parameters``` into separate structs, I think @fweik has proposed this already

Reply:
this should probably be addressed in a dedicated PR

Reply:
@KaiSzuttor you are confusing this with the non-bonded parameters.